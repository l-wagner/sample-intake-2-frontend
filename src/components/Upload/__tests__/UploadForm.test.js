import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestUploadForm from '../UploadForm'

import { initialFormStateMock } from '../../../mocks'

const UploadForm = props => <TestUploadForm form={initialFormStateMock} />

describe('Render UploadForm', () => {
  it('render UploadForm correctly', () => {
    const UploadFormComponent = renderer.create(<UploadForm />).toJSON()
    expect(UploadFormComponent).toMatchSnapshot()
  })
})

