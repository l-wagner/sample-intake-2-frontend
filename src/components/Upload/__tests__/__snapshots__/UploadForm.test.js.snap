// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadForm render UploadForm correctly 1`] = `
<div
  className="UploadForm-form-2"
>
  <form
    className="UploadForm-container-1"
  >
    <div
      className="Dropdown-textField-7"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-8 MuiFormControl-fullWidth-11"
        >
          <label
            className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
            data-shrink={false}
            htmlFor="downshift-0-input"
            id="downshift-0-label"
          >
            Missing translationId: header.material_label for language: \${ languageCode }
            <span
              className="MuiFormLabel-asterisk-29"
            >
               *
            </span>
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedEnd-48"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedEnd-59"
              disabled={false}
              id="material"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-60 MuiInputAdornment-positionEnd-63"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-71 MuiIconButton-root-65"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-70"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-74"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-154"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
          >
            Missing translationId: header.material_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-7"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-8 MuiFormControl-fullWidth-11"
        >
          <label
            className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
            data-shrink={false}
            htmlFor="downshift-1-input"
            id="downshift-1-label"
          >
            Missing translationId: header.application_label for language: \${ languageCode }
            <span
              className="MuiFormLabel-asterisk-29"
            >
               *
            </span>
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-1-label"
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedEnd-48"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedEnd-59"
              disabled={false}
              id="application"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-60 MuiInputAdornment-positionEnd-63"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-71 MuiIconButton-root-65"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-70"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-74"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-154"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
          >
            Missing translationId: header.application_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-7"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-2-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-8 MuiFormControl-fullWidth-11"
        >
          <label
            className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
            data-shrink={false}
            htmlFor="downshift-2-input"
            id="downshift-2-label"
          >
            Missing translationId: header.container_label for language: \${ languageCode }
            <span
              className="MuiFormLabel-asterisk-29"
            >
               *
            </span>
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-2-label"
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedEnd-48"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedEnd-59"
              disabled={false}
              id="container"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-60 MuiInputAdornment-positionEnd-63"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-71 MuiIconButton-root-65"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-70"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-74"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-154"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
          >
            Missing translationId: header.container_helptext for language: \${ languageCode } (2 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-7"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-3-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-8 MuiFormControl-fullWidth-11"
        >
          <label
            className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
            data-shrink={false}
            htmlFor="downshift-3-input"
            id="downshift-3-label"
          >
            Missing translationId: header.species_label for language: \${ languageCode }
            <span
              className="MuiFormLabel-asterisk-29"
            >
               *
            </span>
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-3-label"
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedEnd-48"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedEnd-59"
              disabled={false}
              id="species"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-60 MuiInputAdornment-positionEnd-63"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-71 MuiIconButton-root-65"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-70"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-74"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-154"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
          >
            Missing translationId: header.species_helptext for language: \${ languageCode }
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-7"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-4-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-8 MuiFormControl-fullWidth-11"
        >
          <label
            className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
            data-shrink={false}
            htmlFor="downshift-4-input"
            id="downshift-4-label"
          >
            Missing translationId: header.patient_id_format_label for language: \${ languageCode }
            <span
              className="MuiFormLabel-asterisk-29"
            >
               *
            </span>
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-4-label"
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedEnd-48"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedEnd-59"
              disabled={false}
              id="patient_id_format"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-60 MuiInputAdornment-positionEnd-63"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-71 MuiIconButton-root-65"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-70"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-74"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-154"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
          >
            Missing translationId: header.patient_id_format_helptext for language: \${ languageCode }
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-8 UploadForm-textField-3"
    >
      <label
        className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20"
        data-shrink={false}
        htmlFor="number_of_samples"
      >
        Missing translationId: header.sample_number_label for language: \${ languageCode }
        <span
          className="MuiFormLabel-asterisk-29"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
        onClick={[Function]}
      >
        <input
          aria-describedby="number_of_samples-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputType-56 MuiInput-inputType-41"
          disabled={false}
          id="number_of_samples"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="number"
        />
      </div>
      <p
        className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
        id="number_of_samples-helper-text"
      >
        Missing translationId: header.sample_number_helptext for language: \${ languageCode }
      </p>
    </div>
    <div
      className="MuiFormControl-root-8 UploadForm-textField-3"
    >
      <label
        className="MuiFormLabel-root-23 MuiFormLabel-required-28 MuiInputLabel-required-16 MuiInputLabel-root-12 MuiInputLabel-formControl-17 MuiInputLabel-animated-20 MuiInputLabel-shrink-19"
        data-shrink={true}
        htmlFor="igo_request_id"
      >
        Missing translationId: header.igo_request_id_label for language: \${ languageCode }
        <span
          className="MuiFormLabel-asterisk-29"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31 MuiInputBase-adornedStart-47"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root-60 MuiInputAdornment-positionStart-62"
        >
          <p
            className="MuiTypography-root-92 MuiTypography-body1-101 MuiTypography-colorTextSecondary-125"
          >
            IGO-
          </p>
        </div>
        <input
          aria-describedby="igo_request_id-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-inputAdornedStart-58"
          disabled={false}
          id="igo_request_id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </div>
      <p
        className="MuiFormHelperText-root-83 MuiFormHelperText-required-90"
        id="igo_request_id-helper-text"
      >
        helptext
      </p>
    </div>
  </form>
  <button
    className="MuiButtonBase-root-71 MuiButton-root-128 MuiButton-contained-139 MuiButton-containedSecondary-141 MuiButton-raised-142 MuiButton-raisedSecondary-144 UploadForm-button-6"
    disabled={false}
    form="my-form-id"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-129"
    >
      Missing translationId: header.generate_button for language: \${ languageCode }
    </span>
    <span
      className="MuiTouchRipple-root-154"
    />
  </button>
</div>
`;
